### PR TITLE
Build the applications provided by the container when  the image is created

### LINKED ISSUE #7: Docker-Image lässt sich nicht mehr bauen
Der Build des Images schlägt fehl, folgende Meldung:

```
gpg: no valid OpenPGP data found.
The command '/bin/sh -c /scripts/detect-apt-proxy.sh $APT_PROXY_PORT && apt-get update -y && mkdir /mnt/config && apt-get install -y software-properties-common iptables curl iproute2 ifupdown iputils-ping git apt-transport-https gnupg2 && echo "deb http://deb.debian.org/debian/ unstable main" > /etc/apt/sources.list.d/unstable-wireguard.list && echo "deb http://dl.ffm.freifunk.net/debian-packages/sid main" > /etc/apt/sources.list.d/ffffm_babel.list && printf 'Package: babeld\nPin: release a=sid\nPin-Priority: 150\n' > /etc/apt/preferences.d/limit-babeld_ffffm_sid && printf 'Package: *\nPin: release a=unstable\nPin-Priority: 150\n'> /etc/apt/preferences.d/limit-unstable && curl https://dl.ffm.freifunk.net/info@wifi-frankfurt.de.gpg.pub | apt-key add - && apt-get update -y && apt-get install -y wireguard babeld wg-broker l3roamd mmfd && mkdir -p /etc/iproute2/rt_tables.d && mkdir -p /etc/wg-broker && echo "10    netz" > /etc/iproute2/rt_tables.d/babel.conf && echo "11    l3roamd" >> /etc/iproute2/rt_tables.d/babel.conf && echo "12    babeld" >> /etc/iproute2/rt_tables.d/babel.conf && apt-get install -ystrace libjson-c3' returned a non-zero code: 2
```

### PR DESCRIPTION
**Pull request by penguineer**
Instead of relying on external sources build the applications when the container is created.

This fixes #7

### CODE REVIEW

**Comment by penguineer** [comment]
@christf Bitte mal testen, ob das so läuft.

**Comment by christf** [comment]
it builds and looks ok locally.  I like the multi-staged builds and the simplification by removing apt proxy
Maybe this is now the time when we re-purpose our 2nd batman gateway to a babel gateway so we have again 2 to work with.

**Comment by penguineer** [comment]
> it builds and looks ok locally.

Did you run it? I saw that the wg-broker script did not find wg, but it might be due to my setup?

> I like the multi-staged builds and the simplification by removing apt proxy 

Thanks :)

> Maybe this is now the time when we re-purpose our 2nd batman gateway to a babel gateway so we have again 2 to work with.

We need to change Babel upstream first, this uses the 2nd batman gateway.

**Comment by christf** [comment]
just a question: how do you use your apt cache in that setup?

**Comment by penguineer** [comment]
> just a question: how do you use your apt cache in that setup?

I don't. The complexity introduced into the CI system is not worth the gain, especially since we are not layering the image build.

I would rather consider doing some availability/version check in the run script to avoid installing software when it is not needed.

File: README.md [hunk 1]
```diff
@@ -19,12 +19,8 @@ package feed. This allows source-specific routes being transported properly.
 The build can make use of an apt cache.
```
Comment (24367dfa) [inline README.md]: is this still accurate?

**Review by 24367dfa — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (penguineer) [reply README.md]: It is not, thank you for pointing that out.

I have also remove the reference to a specific babeld version.

**Review by 24367dfa — APPROVED** [review]
lgtm

**Pull request merged**